R/plot_methods.R

```diff
@@ -0,0 +1,44 @@
+#' @importFrom ggplot2 autoplot .data
+#' @export
+ggplot2::autoplot
+
+
+#' Plot a summrt_summary object
+#'
+#' @param object A `summrt_summary` object
+#' @param color A character string giving a color
+#' @param add_reference_line Logical. Display a horizontal line at 1.
+#' @param ... not used.
+#'
+#' @return A [ggplot2::ggplot()] object.
+#' @export
+#'
+#' @examples
+#' ex <- readRDS(system.file(
+#'   "extdata", "EpiEstim_example.rds", package = "summrt"
+#' ))
+#' summ_ex <- summarize_rtestimate(ex)
+#' autoplot(summ_ex)
+#' autoplot(summ_ex) + ggplot2::coord_cartesian(ylim = c(0, 2))
+autoplot.summrt_summary <- function(
+    object, color = "dodgerblue4", add_reference_line = TRUE, ...
+) {
+  dat <- as_tibble(object)
+  level <- ""
+  if (!is.null(object$level)) {
+    level <- paste0(sprintf("%02.0f", object$level), "% ")
+  }
+  ylabel <- paste0("Estimated Rt w/ ", level, "confidence band")
+  plot_obj <- ggplot2::ggplot(dat, ggplot2::aes(.data$date)) +
+    ggplot2::geom_line(ggplot2::aes(y = .data$median), col = color) +
+    ggplot2::geom_ribbon(ggplot2::aes(ymin = .data$lb, ymax = .data$ub),
+                         fill = color, alpha = .2) +
+    ggplot2::theme_bw() +
+    ggplot2::ggtitle(label = object$package) +
+    ggplot2::ylab(ylabel)
```

Comment:
I don't actually feel strongly about this at all, but should we set the default to log scale? E.g. (log10 bc it prints better)
```suggestion
    ggplot2::ylab(ylabel) +
    ggplot2::scale_y_continuous(trans = "log10") 
```

Reply:
I'd kind of lean toward `raw` scale. The values should generally be near 1, so a log transform doesn't do much. On the other hand, it can be confusing to some who aren't used to looking at log-scaled plots. Since the function returns the ggplot object, the user can always add the transformation if they want it (though removing it if we do it would print an annoying message).

I'd lean toward a function argument to make it optional, or not at all. Happy to do either!

Reply:
I think I agree that because the function returns a ggplot object, the user can choose to log scale if they prefer, but better to not include as a default